### DISCUSSION TITLE
Is there a way to mix lf Internal commands with bash commnads?

### DISCUSSION THREAD

**Discussion by Archie-2021** [gokcehan/lf#1267] [General]
For example I like to use a fuzzy finder like fzf or dmenu with some of my commands , say for example `cd` (the internal lf cd not bash cd). How can I mix Bash and Internal commands together. If possible that could open up some cool things to happen. 

Thanks

**Comment by joelim-work** [comment]
You can define custom `lf` commands in your config file which run shell commands (put them inside `${{ }}`), and inside shell commands you can use `lf -remote "send $id <command>"` to trigger `lf` commands.

Here is a basic example that uses the shell to create a directory and then invoke the internal `select` command to select it:

```shell
# usage `:mkdir foo`
cmd mkdir ${{
    mkdir "$1"
    lf -remote "send $id select $1"
}}
```

If you look at the [Tips](https://github.com/gokcehan/lf/wiki/Tips) and [Integrations](https://github.com/gokcehan/lf/wiki/Integrations) section of the wiki, many people have contributed useful configuration examples, for instance [how to integrate](https://github.com/gokcehan/lf/wiki/Integrations#fzf) with `fzf`.

**Reply by Archie-2021** [reply]
> You can define custom `lf` commands in your config file which run shell commands (put them inside `${{ }}`), and inside shell commands you can use `lf -remote "send $id <command>"` to trigger `lf` commands.
> 
> Here is a basic example that uses the shell to create a directory and then invoke the internal `select` command to select it:
> 
> ```shell
> # usage `:mkdir foo`
> cmd mkdir ${{
>     mkdir "$1"
>     lf -remote "send $id select $1"
> }}
> ```
> 
> If you look at the [Tips](https://github.com/gokcehan/lf/wiki/Tips) and [Integrations](https://github.com/gokcehan/lf/wiki/Integrations) section of the wiki, many people have contributed useful configuration examples, for instance [how to integrate](https://github.com/gokcehan/lf/wiki/Integrations#fzf) with `fzf`.

Awesome, thank you very much for this great info.
